File: 23-Java-Kotlin-Scala-Swift/23-Java-Kotlin-Scala-Swift/OrderProcessor.scala
case class Order(id: String, amount: Double)

object OrderProcessor {
  // Scala IO: Referential transparency & Functional purity
  def processOrder(order: Order): IO[Unit] = 
    IO {
      val converted = order.amount * 1.1
      println(s"Scala processed: ${order.id} -> $converted")
    }
}

File: 23-Java-Kotlin-Scala-Swift/23-Java-Kotlin-Scala-Swift/OrderProcessor.kt
data class Order(val id: String, val amount: Double)

class OrderProcessor {
    // Kotlin Coroutines: Suspension-based concurrency
    suspend fun processOrder(order: Order) {
        val converted = order.amount * 1.1
        println("Kotlin processed: ${order.id} -> $converted")
    }
}
